BlockBlock/Assets/Scripts/GameManager.cs

```diff
@@ -1,359 +1,374 @@
+                        // Move block to position and check if it is a valid position
+                        block.transform.position = new Vector2(x, y);
+                        RoundPosition(block.transform);
+
+                        if (IsValidPosition(block))
+                        {
+                            // Reset block back to its spawn position
+                            block.ResetPosition();
+
+                            return true;
+                        }
+                    }
+                }
+                
+                // Reset block back to its spawn position after checking
+                block.ResetPosition();
+            }
+        }
+
+        return false;
+    }
+
+    private void ClearLines()
+    {
+        // Used to keep track of which rows/cols to clear
+        HashSet<int> rowsToClear = new HashSet<int>();
+        HashSet<int> colsToClear = new HashSet<int>();
+
+        // Check for full rows
+        foreach (int row in rowsToCheck)
+        {
+            if (IsLineFull(row, true))
+            {
+                rowsToClear.Add(row);
+            }
+        }
+
+        // Check for full cols
+        foreach(int col in colsToCheck)
+        {
+            if (IsLineFull(col, false))
+            {
+                colsToClear.Add(col);
+            }
+        }
+
+        // Clear rows
+        foreach(int row in rowsToClear)
+        {
+            ClearLine(row, true);
+        }
+
+        // Clear cols
+        foreach(int col in colsToClear)
+        {
+            ClearLine(col, false);
+        }
+
+        // Clear hashsets used to indicate which rows/cols to check
+        rowsToCheck.Clear();
+        colsToCheck.Clear();
+    }
+
+    // Clears row/col
+    private void ClearLine(int index, bool rowFlag)
+    {
+        // Clear row if flag is set
+        if (rowFlag)
+        {
+            for (int x = 0; x < GRID_WIDTH; x++)
+            {
+                if (grid[x, index] != null)
+                {
+                    // set trigger to transition to the destroy animation
+                    grid[x, index].gameObject.GetComponent<Animator>().SetTrigger("DestroyAnimation");
+
+                    // wait 1 second to allow the animation to finish before destroying the block piece
+                    Destroy(grid[x, index].gameObject, 1f);
+                    grid[x, index] = null;
+                }
+            }
+        }
+        // Otherwise clear col
+        else
+        {
+            for (int y = 0; y < GRID_HEIGHT; y++)
+            {
+                if (grid[index, y] != null)
+                {
+                    // set trigger to transition to the destroy animation
+                    grid[index, y].gameObject.GetComponent<Animator>().SetTrigger("DestroyAnimation");
+
+                    // wait 1 second to allow the animation to finish before destroying the block piece
+                    Destroy(grid[index, y].gameObject, 1f);
```

Comment:
For the 1f, instead of using the constant, maybe get the animation time from the AnimationClip.length (not sure if that accessible):  https://docs.unity3d.com/ScriptReference/AnimationClip-length.html

Reply:
Something like grid[index, y].GetComponent<Animator>().GetCurrentAnimatorClipInfo(0).length (at this point, it looks like grid[index, y].gameObject gets repeated a lot so maybe save that in a var